src/Analyzer/Pass/Statement/ConstantNaming.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace PHPSA\Analyzer\Pass\Statement;
+
+use PhpParser\Node\Stmt\ClassConst;
+use PHPSA\Analyzer\Pass\AnalyzerPassInterface;
+use PHPSA\Analyzer\Pass\ConfigurablePassInterface;
+use PHPSA\Context;
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+
+class ConstantNaming implements ConfigurablePassInterface, AnalyzerPassInterface
+{
+
+    /**
+     * @param ClassConst $stmt
+     * @param Context $context
+     * @return bool
+     */
+    public function pass(ClassConst $stmt, Context $context)
+    {
+        foreach ($stmt->consts as $const) {
+            if ($const->name != strtoupper($const->name)) {
+                $context->notice(
+                    'constant.naming',
+                    'Constant names should be all uppercase.',
+                    $stmt
+                );
+
+                return true;
```

Comment:
If you return inside the loop, all the constants might not be analyzed. Take the following code for instance, only the first invalid constant will be reported:

```php
class Foo {
    const Bar = 1, Baz = 2;
}
```